addons/portal/views/portal_templates.xml

```diff
@@ -842,34 +842,40 @@
             <h3>Your Account Address</h3>
-            <t t-call="portal.address_card"/>
-            <h5>Other Addresses</h5>
-            <ul>
+            <t t-call="portal.address_card">
+                <t t-set="_classes" t-value="'w-md-50 bg-primary'"/>
+            </t>
+            <h4 class="mt32">Other Addresses</h4>
+            <ul class="list-unstyled d-flex flex-wrap gap-3">
                 <t t-foreach="partner.child_ids" t-as="partner">
-                    <t t-call="portal.address_card"/>
+                    <t t-call="portal.address_card">
+                        <t t-set="_classes" t-value="'w-md-auto'"/>
+                    </t>
                 </t>
             </ul>
-            <a href="/my/addresses/new" class="btn btn-primary justify-content-start gap-3 align-items-center h-100 p-3">Add an address</a>
+            <a href="/my/addresses/new" class="btn btn-primary justify-content-start align-items-center gap-3 h-100">Add an address</a>
         </t>
     </template>
 
     <template id="address_card">
-        <a t-attf-href="/my/account?address={{partner.id}}" t-att-title="title" class="d-flex justify-content-start gap-3 align-items-center h-100 p-3">
-            <div>
-                <h5 class="mb-0 d-flex gap-3">
-                    <i t-if="partner.type == 'invoice'" class="fa fa-file-text-o fa-fw" role="img"/>
-                    <i t-if="partner.type == 'delivery'" class="fa fa-truck fa-fw" role="img"/>
-                    <span t-out="partner.name"/>
+        <div t-attf-class="card w-100 {{ _classes }}">
+            <div class="card-body">
+                <a t-attf-href="/my/account?address={{partner.id}}" t-att-title="title" class="float-end ms-2">
+                    <i class="fa fa-pencil me-1"/>
+                    Edit
+                </a>
+                <h5>
+                    <i t-if="partner.type == 'invoice'" class="fa fa-fw fa-file-text-o text-primary" role="img"/>
+                    <i t-if="partner.type == 'delivery'" class="fa fa-truck fa-fw text-primary" role="img"/>
+                    <i t-if="partner.type == 'contact'" class="fa fa-address-card-o fa-fw text-primary" role="img"/>
+                    <t t-out="partner.name"/>
                 </h5>
-
-                <span>
-                    <t t-if="partner.email" t-out="partner.email"/>
-                    <div>
-                        <t t-if="partner.zip" t-out="partner.zip + ' '"/><t t-out="partner.city"/>
-                    </div>
-                    <t t-if="partner.state_id" t-out="partner.state_id.name + ' '"/><t t-out="partner.country_id.name"/>
-                </span>
+                <div class="d-flex flex-column">
+                    <span t-if="partner.email" t-out="partner.email"/>
+                    <span t-if="partner.zip" t-out="partner.zip + ' ' + partner.city"/>
+                    <span t-if="partner.state_id" t-out="partner.state_id.name + ' ' + partner.country_id.name"/>
```

Comment:
Are we sure we don't want to show the city without zip or the country without state?

Reply:
Yes, we use the same logic as for the addresses in product checkout
<img alt="image" src="https://github.com/odoo/odoo/assets/128030743/4b4a7a04-72be-44ee-b66e-b8a162240388">